bot/kodiak/refresh_pull_requests.py

```diff
@@ -0,0 +1,199 @@
+
+
+async def refresh_pull_requests_for_installation(
+    *, installation_id: str, redis: RedisConnection
+) -> None:
+    login = await get_login_for_install(installation_id=installation_id)
+    token = await get_token_for_install(installation_id=installation_id)
+    res = await http.post(
+        "https://api.github.com/graphql",
+        json=dict(query=QUERY, variables=dict(login=login)),
+        headers=dict(Authorization=f"Bearer {token}"),
+    )
+    res.raise_for_status()
+
+    organizationdata = res.json()["data"]["organizationdata"]
+    userdata = res.json()["data"]["userdata"]
+    user_kind = None
+    if organizationdata is not None:
+        user_kind = "organization"
+        data = organizationdata
+    elif userdata is not None:
+        user_kind = "user"
+        data = userdata
+    else:
+        raise ValueError("missing data for user/organization")
+
+    events = []
+    for repository in data["repositories"]["nodes"]:
+        repo_name = repository["name"]
+        for pull_request in repository["pullRequests"]["nodes"]:
+            events.append(
+                WebhookEvent(
+                    repo_owner=login,
+                    repo_name=repo_name,
+                    pull_request_number=pull_request["number"],
+                    installation_id=installation_id,
+                )
+            )
+    for event in events:
+        await redis.zadd(
+            event.get_webhook_queue_name(),
+            {event.json(): time.time()},
+            only_if_not_exists=True,
+        )
+    logger.info(
+        "pull_requests_refreshed",
+        installation_id=installation_id,
+        events_queued=len(events),
+        user_kind=user_kind,
+    )
+
+
+class RefreshPullRequestsMessage(BaseModel):
+    installation_id: str
+
+
+async def main_async() -> None:
+    redis_db = 0
+    try:
+        redis_db = int(conf.REDIS_URL.database)
+    except ValueError:
+        pass
+    redis = await asyncio_redis.Connection.create(
+        host=conf.REDIS_URL.hostname or "localhost",
+        port=conf.REDIS_URL.port or 6379,
+        password=conf.REDIS_URL.password or None,
+        db=redis_db,
+    )
+    while True:
+        try:
+            res: BlockingPopReply = await redis.blpop(
+                ["kodiak:refresh_pull_requests_for_installation"], timeout=5
+            )
+        except asyncio_redis.exceptions.TimeoutError:
+            logger.info("pull_request_refresh", timeout_reached=True)
+            continue
+        pr_refresh_message = RefreshPullRequestsMessage.parse_raw(res.value)
+        installation_id = pr_refresh_message.installation_id
+        start = time.time()
+        await refresh_pull_requests_for_installation(
+            installation_id=installation_id, redis=redis
+        )
+        logger.info(
+            "pull_request_refresh",
+            installation_id=installation_id,
+            duration=time.time() - start,
```

Comment:
```suggestion
            duration=time.monotonic() - start,
```

Reply:
Monotonic isn't what you think it is. It's like a program counter, it starts from the start of the program.

```
❯ python3 -c 'import time; print(time.monotonic())'
0.039981657
```